hcn/hcn.go

```diff
@@ -93,12 +94,12 @@ type HostComputeQuery struct {
 }
 
 type ExtraParams struct {
-	Resources        interface{} `json:",omitempty"`
-	SharedContainers interface{} `json:",omitempty"`
-	LayeredOn        string      `json:",omitempty"`
-	SwitchGuid       string      `json:",omitempty"`
-	UtilityVM        string      `json:",omitempty"`
-	VirtualMachine   string      `json:",omitempty"`
+	Resources        json.RawMessage `json:",omitempty"`
+	SharedContainers json.RawMessage `json:",omitempty"`
```

Comment:
just curious: why did you change these?

Reply:
Easier to unmarshal into something as it's just bytes instead of an interface

Reply:
```go
var exi ExtraInfo
if err := json.Unmarshal(ep.Health.Extra.Resources, &exi); err != nil {
	return errors.Wrapf(err, "failed to unmarshal resource data from endpoint %q", endpointID)
}
```

Reply:
ah, good point